documentation/tutorial.md

```diff
@@ -68,3 +71,416 @@ tutorial overlord    localhost/localhost  2.9.37   unsupported  23:20:53Z
+
+   The monitoring data will be available on a MongoDB website with a unique URL accessible to you
+   and anyone you share the URL with. MongoDB may use this information to make product
+   improvements and to suggest MongoDB products and deployment options to you.
+
+   To enable free monitoring, run the following command: db.enableFreeMonitoring()
+   To permanently disable this reminder, run the following command: db.disableFreeMonitoring()
+------
+
+mongodb [primary] admin>
+```
+
+You can now interact with MongoDB directly using any [MongoDB commands](https://www.mongodb.com/docs/manual/reference/command/). For example entering `show dbs` should output something like:
+```
+admin   172.00 KiB
+config  120.00 KiB
+local   404.00 KiB
+```
+
+Feel free to test out any other MongoDB commands, when you’re ready to leave the shell you can just type `exit`
+
+
+## Scaling Charmed MongoDB: 
+Replication is a popular feature of MongoDB; replicas copy data making a database highly reliable. 
+
+
+### Add replicas:
+You can add two replicas to your deployed MongoDB application with:
+```
+juju add-unit mongodb -n2
+```
+
+You can now watch the replica set add these replicas with: `watch -c juju status --color`. You’ll know that all three replicas are ready when `watch -c juju status --color` reports:
+```
+Every 2.0s: juju status --color                                                                                                                                                ip-172-31-11-104: Fri Dec  2 14:36:50 2022
+
+Model     Controller  Cloud/Region         Version  SLA          Timestamp
+tutorial  overlord    localhost/localhost  2.9.37   unsupported  14:42:04Z
+
+App      Version  Status  Scale  Charm    Channel   Rev  Exposed  Message
+mongodb           active      3  mongodb  dpe/edge   96  no       Replica set primary
+
+Unit        Workload  Agent  Machine  Public address  Ports      Message
+mongodb/0*  active    idle   0        10.23.62.156    27017/tcp  Replica set primary
+mongodb/1   active    idle   1        10.23.62.55     27017/tcp  Replica set secondary
+mongodb/2   active    idle   2        10.23.62.243    27017/tcp  Replica set secondary
+
+Machine  State    Address       Inst id        Series  AZ  Message
+0        started  10.23.62.156  juju-d35d30-0  focal       Running
+1        started  10.23.62.55   juju-d35d30-1  focal       Running
+2        started  10.23.62.243  juju-d35d30-2  focal       Running
+```
+
+You can trust that Charmed MongoDB added these replicas correctly. But if you wanted to verify the replicas got added correctly you could connect to MongoDB via `mongosh`. Since your replica set has 2 additional hosts you will need to update the hosts in your URI. Using the IP addresses from `juju status` add all of your hosts to the URI and separate them with commas. Your URI should look like this:
+```
+mongodb://<username>:<password>@<host1,<host2>,<host3>>/<database name>?replicaSet=<replica set name>
```

Comment:
Uhm, just wondering here... Do we have a way to provide this information to the user out of the box? Like an action that would return this info. It is not something to be worked on in this task, but I thought it was just worth mentioning that this is something nice to provide to the user, if not already there (in this case I would put how to do this in the tutorial...)

Reply:
GREAT question. IMO from a **strict** MongoDB point of view those instructions are the **only way** to verify. (other than writing data to the primary replica and checking it copied on the secondary) 

But if we are being less strict, the user could verify this by typing `juju status` and seeing:
```
Model     Controller  Cloud/Region         Version  SLA          Timestamp
tutorial  overlord    localhost/localhost  2.9.37   unsupported  10:30:16Z

App                  Version  Status  Scale  Charm                Channel   Rev  Exposed  Message
mongodb                       active      2  mongodb              dpe/edge   96  no       Replica set secondary

Unit                    Workload  Agent  Machine  Public address  Ports      Message
mongodb/0*              active    idle   0        10.23.62.156    27017/tcp  Replica set secondary
mongodb/1               active    idle   1        10.23.62.55     27017/tcp  Replica set primary

Machine  State    Address       Inst id        Series  AZ  Message
0        started  10.23.62.156  juju-d35d30-0  focal       Running
1        started  10.23.62.55   juju-d35d30-1  focal       Running
```
(however messages such as `replica set primary` and `replica set secondary` only appear once the update status hook is triggered which can take up to 5 minutes. (unless the hook is triggered to fire quicker -- which is easy to do)

Reply:
so, there is no action that returns the uri? I suppose one could do the "juju show-unit" to see the databag though. It is a bit hacky, but figuring this out all the time bit by bit (username, passwords, hosts, replica_set name) can be a bit tedious. It is good for the tutorial, but it would also be good to have either an action or using the show-unit, as a tip